src/prefect/agent/docker/agent.py

```diff
@@ -518,20 +518,27 @@ def populate_env_vars(
         else:
             api = config.cloud.api
 
-        env = {}
+        env = {}  # type: ignore
         # Populate environment variables, later sources overriding
-
-        # 1. Logging config (optional)
-        # We only set this on non-DockerRun runs for backwards compatibility.
-        # In the future we don't want the agent to set logging config on a flow
-        # run unless explicitly asked.  We do this early on so later config
-        # sources can override
-        if run_config is None:
-            env.update({"PREFECT__LOGGING__LEVEL": config.logging.level})
-
-        # 2. Values set on the agent via `--env`
+        # 1. Values set on the agent via `--env`
         env.update(self.env_vars)
 
+        # 2. Logging config - if logging level is passed as an environment
+        # variable on the run config, it should take precedence
+        existing_logging_level = env.get(
+            "PREFECT__LOGGING__LEVEL", config.logging.level
+        )
+        if run_config is not None and run_config.env is not None:
+            env.update(
+                {
+                    "PREFECT__LOGGING__LEVEL": run_config.env.pop(
+                        "PREFECT__LOGGING__LEVEL", existing_logging_level
+                    )
+                }
+            )
+        else:
+            env.update({"PREFECT__LOGGING__LEVEL": existing_logging_level})
```

Comment:
Won't the top half of this occur on L544 with `env.update(run_config.env)` ? It seems like you could drop this entire if/else and just do `env.update({"PREFECT__LOGGING__LEVEL": existing_logging_level})` which will be overridden in the next block if it exists

Reply:
lol

Reply:
that is roughly how I had it, Chris suggested making this logic more explicit because we're treating `PREFECT__LOGGING__LEVEL` in a first class way unlike other environment variables

Reply:
I find this block far more complicated and we're not actually treating it differently here. The run config env overrides all other values regardless of whether it's logging related or not.

Reply:
@cicdw thoughts here? It also seems to me like these efforts are making things much more complicated